wagtail/snippets/views/snippets.py

```diff
@@ -747,17 +780,29 @@ class SnippetViewSet(ModelViewSet):
     #: The template to use for the history view.
     history_template_name = ""
 
-    def __init__(self, model, **kwargs):
-        self.model = model
-        self.app_label = self.model._meta.app_label
-        self.model_name = self.model._meta.model_name
+    def __init__(self, model=None, **kwargs):
+        # Allow model to be defined on the class, or passed in via the constructor
+        self.model = model or self.model
```

Comment:
If the former (setting `model` on the class) is the preferred way, we could also start deprecating the other approach

Reply:
Let's leave both options open for now, and see what gains traction among Wagtail devs. Defining `model` on the class does feel more natural to me as an API, but for people who are used to defining `register_snippet` just on the model, introducing the viewset class as an optional extra might be a dramatic enough change already to start pushing something else as the one true way :-)